indico/modules/events/registration/util.py

```diff
@@ -1168,3 +1169,80 @@ class CustomTicketCode:
+    column_names = {
+        'reg_date': 'Registration Date',
+        'state': 'Registration State',
+        'price': 'Price',
+        'checked_in': 'Checked in',
+        'checked_in_date': 'Check-in Date',
+        'tags_present': 'Tags',
+    }
+
+    headers = ['ID', 'Name']
+
+    dynamic_form_items = []
+    accommodation_col_counter = 0
+
+    for item in display:
+        if item.input_type == 'accommodation':
+            accommodation_col_counter += 1
+            headers.append(item.title)
+            headers.append('Arrival date')
+            headers.append('Departure date')
+        else:
+            headers.append(item.title)
+
+        dynamic_form_items.append(item)
+
+    headers.extend(column_names[item_id] for item_id in static_items if item_id in column_names)
+
+    rows = []
+    for registration in reglist:
+        row = [f'#{registration.friendly_id}', registration.display_full_name]
+        reg_data_by_field = registration.data_by_field
+        for item in dynamic_form_items:
+            reg_data = reg_data_by_field.get(item.id)
+            friendly_data = reg_data.friendly_data if reg_data else None
+
+            if item.input_type == 'accommodation':
+                if friendly_data:
+                    row.append(escape(friendly_data.get('choice', '')))
+                    row.append(format_date(friendly_data.get('arrival_date', '')))
+                    row.append(format_date(friendly_data('departure_date', '')))
+                else:
+                    row.extend(['', '', ''])
+            elif item.input_type == 'multi_choice':
+                multi_choice_data = ', '.join(friendly_data) if friendly_data else ''
+                row.append(escape(multi_choice_data))
+            elif item.input_type == 'sessions':
+                sessions_data = '; '.join(friendly_data) if friendly_data else ''
+                row.append(escape(sessions_data))
+            else:
+                cell_data = str(friendly_data) if friendly_data is not None else ''
+                row.append(escape(cell_data))
+
+        for item_id in static_items:
+            cell_data = ''
+            if item_id == 'reg_date':
+                cell_data = format_datetime(registration.submitted_dt)
+            elif item_id == 'state':
+                cell_data = registration.state.title
+            elif item_id == 'price':
+                cell_data = registration.render_price()
+            elif item_id == 'checked_in':
+                cell_data = 'Yes' if registration.checked_in else 'No'
+            elif item_id == 'checked_in_date':
+                if registration.checked_in_dt:
+                    cell_data = format_datetime(registration.checked_in_dt)
+            elif item_id == 'tags_present':
+                if registration.tags:
+                    cell_data = ', '.join(sorted(t.title for t in registration.tags))
+                    cell_data = escape(cell_data)
```

Comment:
Escaping data should not be necessary anywhere, and certainly not using the legacy escape function. Jinja templates have autoescaping enabled.